executor/native_linux_executor.cpp

```diff
@@ -717,54 +729,29 @@ ExitStatusFeedback NativeLinuxExecutor::GetExitStatusFeedback() {
     return ExitStatusFeedback(last_exit_reason, last_signal);
 }
 
+bool NativeLinuxExecutor::IsFeedbackLocked() {
+    return (lock.use_count() > 1)
```

Comment:
Is this `NativeLinuxExecutor::lock` still used? If not, I think we can remove it.

Reply:
The same for other executors.

Reply:
The `NativeLinuxExecutor::lock` is still used by `GetStdOut()` and `GetStdErr()`. If `stdout_buffer` and `stderr_buffer` (`std::vector<std::uint8_t>`) can have own locks, then we can remove it.

Reply:
Oh sorry I forgot it. Okay let's leave it then.